exchanges/bybit/bybit_wrapper.go

```diff
@@ -724,79 +579,65 @@ func (by *Bybit) UpdateOrderbook(ctx context.Context, p currency.Pair, assetType
-				Currency: currency.NewCode(balances[i].CoinName),
-				Total:    balances[i].Total.Float64(),
-				Hold:     balances[i].Locked.Float64(),
-				Free:     balances[i].Total.Float64() - balances[i].Locked.Float64(),
-			}
-		}
-
-		acc.Currencies = currencyBalance
-
-	case asset.CoinMarginedFutures, asset.USDTMarginedFutures, asset.Futures:
-		balances, err := by.GetFutureWalletBalance(ctx, "")
-		if err != nil {
-			return info, err
-		}
-
-		var i int
-		currencyBalance := make([]account.Balance, len(balances))
-		for coinName, data := range balances {
-			currencyBalance[i] = account.Balance{
-				Currency: currency.NewCode(coinName),
-				Total:    data.WalletBalance,
-				Hold:     data.WalletBalance - data.AvailableBalance,
-				Free:     data.AvailableBalance,
+	case asset.Spot, asset.Options,
+		asset.USDCMarginedFutures,
+		asset.USDTMarginedFutures:
+		switch by.AccountType {
+		case accountTypeUnified:
+			accountType = "UNIFIED"
+		case accountTypeNormal:
+			if assetType == asset.Spot {
+				accountType = "SPOT"
+			} else {
+				accountType = "CONTRACT"
 			}
-			i++
-		}
-
-		acc.Currencies = currencyBalance
-
-	case asset.USDCMarginedFutures:
-		balance, err := by.GetUSDCWalletBalance(ctx)
-		if err != nil {
-			return info, err
-		}
-
-		acc.Currencies = []account.Balance{
-			{
-				Currency: currency.USD,
-				Total:    balance.WalletBalance.Float64(),
-				Hold:     balance.WalletBalance.Float64() - balance.AvailableBalance.Float64(),
-				Free:     balance.AvailableBalance.Float64(),
-			},
 		}
-
+	case asset.CoinMarginedFutures:
+		accountType = "CONTRACT"
 	default:
 		return info, fmt.Errorf("%s %w", assetType, asset.ErrNotSupported)
 	}
+	balances, err := by.GetWalletBalance(ctx, accountType, "")
+	if err != nil {
+		return info, err
+	}
+	currencyBalance := []account.Balance{}
+	reservedPairs := make(map[string]struct{})
+	for i := range balances.List {
+		for c := range balances.List[i].Coin {
+			balance := account.Balance{
+				Currency: currency.NewCode(balances.List[i].Coin[c].Coin),
+				Total:    balances.List[i].TotalWalletBalance.Float64(),
+				Free:     balances.List[i].Coin[c].AvailableToWithdraw.Float64(),
+				Borrowed: balances.List[i].Coin[c].BorrowAmount.Float64(),
+				Hold:     balances.List[i].Coin[c].WalletBalance.Float64() - balances.List[i].Coin[c].AvailableToWithdraw.Float64(),
+			}
+			if assetType == asset.Spot && balances.List[i].Coin[c].AvailableBalanceForSpot.Float64() != 0 {
+				balance.Free = balances.List[i].Coin[c].AvailableBalanceForSpot.Float64()
+			}
+			reservedPairs[balances.List[i].Coin[c].Coin] = struct{}{}
+			currencyBalance = append(currencyBalance, balance)
+		}
+	}
+	acc.Currencies = currencyBalance
 	acc.AssetType = assetType
 	info.Accounts = append(info.Accounts, acc)
-
 	creds, err := by.GetCredentials(ctx)
 	if err != nil {
 		return account.Holdings{}, err
 	}
-	if err := account.Process(&info, creds); err != nil {
+	err = account.Process(&info, creds)
```

Comment:
From this [nit](https://github.com/thrasher-corp/gocryptotrader/pull/1301#discussion_r1376857900) This is still incomplete.